tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles.cc

```diff
@@ -78,6 +78,52 @@ void TrtShapeOptimizationProfile::ImplicitBatchModeCompatibleStrategy(
   }
 }
 
+// Create a new Dims vector where the dimensions are the result of a binary
+// operation between dimensions in lhs and rhs.
+template <typename BinaryOperation>
+Status ShapeProfileBinaryOp(const std::vector<nvinfer1::Dims>& lhs,
+                            const std::vector<nvinfer1::Dims>& rhs,
+                            BinaryOperation op,
+                            std::vector<nvinfer1::Dims>* out) {
```

Comment:
Suggest to make this change:
// For each dimension d, accumulates the dimension value in rhs[d] to lhs[d] to construct shape for shape profiling.
// The BinaryOperation may be std::min, std::max etc.
template <typename BinaryOperation>
Status ShapeProfileBinaryOp(std::vector<nvinfer1::Dims>* lhs,
                            const std::vector<nvinfer1::Dims>& rhs,
                            BinaryOperation op) {

Reply:
I have changed the function to return the results in the first arg, and improved the docstring.